Hull classifiers to antennas

drNovikov opened the pull request:
Hull classifiers changed to be antennas and not beacons. This removes the annoying glow and allows to turn broadcasting off without turning off the classifier itself. 

Downsides: consumes more power, may be hacked in the Autpilot/Radar metagame and give away ships/bases locations to enemy.

Will look for a way to make the default beacons non-glowing and, if possible, separate the broadcasting feature from turning the block on/off. Will try to decrease the new "antenna" classifiers power consumption.

All the surfaces of beacons and hull classifiers made mountable to prevent problems with building (for more convenient building).

zrisher committed:
cp round reward imprvs and fixes

zrisher committed:
add ability to show/hide CP GPS markers

drNovikov committed:
Changed HullClassifier to MyRadioAntenna instead of MyBeacon

drNovikov committed:
Changed HullClassifier to MyRadioAntenna instead of MyBeacon

drNovikov committed:
Changed HullClassifier mount points so they cover the whole surface

drNovikov committed:
Changed HullClassifier mount points so they cover the whole surface

drNovikov committed:
Bacon surfaces made fully mountable

drNovikov committed:
Bacon surfaces made fully mountable

zrisher commented on .gitignore:
```diff
@@ -5,3 +5,4 @@ Data/Scripts/GardenConquest/GardenConquest.v11.suo
 modinfo.sbmi
 GardenConquestTests/bin/
 GardenConquestTests/obj/
+*.bak
```
What file were you working with that had this extension?

drNovikov replied:
*bak is how I marked the backups.

zrisher replied:
Which backups are you referring to? So with git, you never need to save backups of files that have already been committed to the tree, because they're always available through the commit history. You can check out any version of any file from any time it's been committed. 

The gitignore is meant to include all files that should everyone working on this project would want to ignore - so we only include file types that are directly correlated with the body of code. Personal preferences we leave to the global .gitignore file (likely already available in your ~ directory). 

This isn't to be a dick, it's just long term better practice to be semantic about what we include to help describe the project and ensure no one is ever surprised because they try to add a legitimate file but it's ignored due to an unfamiliar ignore.

drNovikov replied:
Understood :) Thanks for the clarification, I've never used github before, so I am still new to the practices.

Will move this to my global .gitignore Sorry for cluttering the file, I thought it influences my github only.

*.bak is how I use Notepad++ to quickly test an idea without involving repositories. I just copy a file to *.bak, and then test ideas by changing the original file and seeing the effect. Reverting is faster than with git.

zrisher commented on Data/CubeBlocks.sbc:
```diff
@@ -2,11 +2,91 @@
+        <TypeId>Beacon</TypeId>
+        <SubtypeId>LargeBlockBeacon</SubtypeId>
+      </Id>
+      <DisplayName>DisplayName_Block_Beacon</DisplayName>
```
What's the purpose of this commented-out xml? How were you attempting to change the beacon that didn't work, and how is that information contained here? If this is highly similar to the other blocks below I'd prefer to just list what we change to adapt to a "modified" beacon instead of repeating it all.

drNovikov replied:
That was to enlarge the default beacon mounting area. Will remove.

zrisher commented:
For the attachment points, not sure how I feel about allowing it to attach everywhere. Our new model is designed with the lower half attachment points in mind and would not actually be touching on top. Plus I think limited attachment points is an interesting building dynamic. Could you please revert those changes? We can open an issue to discuss if this is something we need if you feel strongly about it.

drNovikov commented:
I will open an issue to discuss: https://github.com/stackcollision/GardenConquest/issues/119

Not touching on top could lead to breaking existing designs.

drNovikov committed:
Beacon and Hull Classifiers mount points reverted to vanilla

drNovikov committed:
Beacon and Hull Classifiers mount points reverted to vanilla

drNovikov committed:
Fixed a typo in a comment

drNovikov committed:
Fixed a typo in a comment

zrisher commented:
Looking good, please fix the `.gitignore` if that makes sense and you agree; will test and incorporate this weekend.

drNovikov committed:
Removed my *.bak from .gitignore

drNovikov committed:
Removed my *.bak from .gitignore

drNovikov commented:
Removed my stuff from the .gitignore

drNovikov committed:
"/gc fleet stop" enables thrusters, enables gyroscopes, sets gyro power to 100%, disables gyro overrides

The "/gc fleet stop" command enables thrusters, also enables gyroscopes,
sets their power to 100%, and disables their overrides. This was added
because the command failed to stop ships on the server if their
thrusters were turned off.

drNovikov committed:
"/gc fleet stop" enables thrusters, enables gyroscopes, sets gyro power to 100%, disables gyro overrides

The "/gc fleet stop" command enables thrusters, also enables gyroscopes,
sets their power to 100%, and disables their overrides. This was added
because the command failed to stop ships on the server if their
thrusters were turned off.

zrisher commented:
Per our convo, please rebase to master.

drNovikov committed:
Added ResourceSinkGroup to hull classifier definitions

Added <ResourceSinkGroup>Utility</ResourceSinkGroup> to every Hull
Classifier definition

drNovikov committed:
Added ResourceSinkGroup to hull classifier definitions

Added <ResourceSinkGroup>Utility</ResourceSinkGroup> to every Hull
Classifier definition

drNovikov committed:
Merge remote-tracking branch 'origin/dev-drnovikov' into dev-drnovikov
